### PR TITLE
feat(python): add common user input

### PR DESCRIPTION
**Pull request by elsapet**
## Description

Splitting user input rules into separate PR (so we can move forward without branching off path traversal PR with weird bug)

## Checklist
If this is your first time contributing please [sign the CLA](https://docs.bearer.com/contributing/)

- [ ] My rule has adequate metadata to explain its use.

### CODE REVIEW

File: rules/python/shared/common/user_input.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+imports:
+  - python_shared_django_user_input
+  - python_shared_lang_user_input
+patterns:
+  - pattern: $<PYTHON_SHARED_COMMON_USER_INPUT>
+    filters:
+      - either:
+          - variable: PYTHON_SHARED_COMMON_USER_INPUT
+            detection: python_shared_django_user_input
+            scope: cursor
```
Comment (didroe) [inline rules/python/shared/common/user_input.yml:13]: Doesn't really matter, but you can do this which is slightly better performance-wise
```suggestion
            scope: cursor_strict
```

File: rules/python/shared/common/user_input.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+          - variable: PYTHON_SHARED_COMMON_USER_INPUT
+            detection: python_shared_lang_user_input
+            scope: cursor
```
Comment (didroe) [inline rules/python/shared/common/user_input.yml:16]:
```suggestion
            scope: cursor_strict
```

File: rules/python/shared/lang/dynamic_input.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  - id: python_shared_lang_dynamic_input_parser
+    patterns:
+      - pattern: $<ARG_PARSER>()
+        filters:
+          - variable: ARG_PARSER
+            detection: python_shared_lang_dynamic_input_argument_parser
+  - id: python_shared_lang_dynamic_input_argument_parser
+    patterns:
+      - ArgumentParser # fallback
+      - from argparse import ArgumentParser as $<!>$<_>
```
Comment (elsapet) [inline rules/python/shared/lang/dynamic_input.yml:21]: This isn't matching...

It compiles to the following, which matches as expected on Tree Sitter 🤔 

```
([(import_from_statement  . [(dotted_name . [ (identifier )] @param1 .)]  . [(aliased_import . [(dotted_name . [ (identifier )] @param2 .)]  . alias: (_) @match .)] .)] @root)
```

Comment (elsapet) [reply rules/python/shared/lang/dynamic_input.yml:26]: ^ https://github.com/Bearer/bearer/pull/1592

**Pull request merged**